### DISCUSSION TITLE
problems getting this to work

### DISCUSSION THREAD

**Discussion by shinrax2** [cspiel1/fspctl#4] [Announcements]
ive got a FSP Powermanager 10kW Hybrid with P17 card (i have two variants one for plexlog, one for the smartmeter, i guess the difference is in the firmware)
its connected via a usb - rs485 adapter but i cant get fspctl to work

plexlog card: unknown modbus error
smartmeter card: crc error (i think the smartmeter card is the way to go)

do i need to check the communication settings on the p17 card?

also according to the code of fspctl its setting up for rs232 communication
is a usb - rs485 adapter really the right choice? or do i need to go for the rs232 port of the inverter?

any tip would be greatly appreciated

**Comment by cspiel1** [comment]
The P17 is a RS485 card. Currently I have not more information like already in the README.md. You have to check the DIP switches on the P17 card.
![1702388354494](https://github.com/cspiel1/fspctl/assets/14180877/2c490fe0-5397-4356-8358-5fb5ea668aca)

**Reply by shinrax2** [reply]
yes im aware of that
i will check the DIPs and change config accordingly thanks so far

my confusion comes from [here](https://github.com/cspiel1/fspctl/blob/a823745fa793c4867ea525bcfe1e97fe08e55308/src/main.c#L160)
im not familar with libmodbus but setting RS232 serial mode for a RS485 card seems wrong to me

**Reply by cspiel1** [reply]
Ah yes, this is confusing. Isn't there a Doxygen in the libmodbus? There are two types of modbus I darkly remember:
- Modbus RTU (MODBUS_RTU_RS232) and
- Modbus TCP
The setting is correct and you need a USB RS485 converter.
The README.md is correct and it is working in my setup.
